File: internal/core/domain/ports/payment_topic.go
package ports

import "context"

type PaymentTopic interface {
	PublishPaymentCreationSuccess(ctx context.Context, externalId string, code string) error
	PublishPaymentCreationFailure(ctx context.Context, externalId string, reason string) error
	PublishPaymentStatus(ctx context.Context, externalId string, status string) error
}

File: src/core/domain/ports/payment_topic.ts
export interface IPaymentTopic {
	publishPaymentCreationSuccess(externalId: string, code: string): Promise<void>
	publishPaymentCreationFailure(externalId: string, reason: string): Promise<void>
	publishPaymentStatus(externalId: string, status: string): Promise<void>
}
